server/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -260,6 +260,34 @@ protected int doHashCode() {
 
     @Override
     protected Query doToQuery(SearchExecutionContext context) throws IOException {
+        Query innerQuery;
+        NestedObjectMapper mapper = context.nestedLookup().getNestedMappers().get(path);
```

Comment:
I think all checks that were previously done in `toQuery` before this nested lookup need to occur. 

Including
```
 NestedObjectMapper mapper = context.nestedLookup().getNestedMappers().get(path);
        if (mapper == null) {
            if (ignoreUnmapped) {
                return new MatchNoDocsQuery();
            } else {
                throw new IllegalStateException("[" + NAME + "] failed to find nested object under path [" + path + "]");
            }
        }
```

And
```
if (context.allowExpensiveQueries() == false) {
```

Reply:
Totally - I moved the meat out of the sandwich. I have corrected it now.